File: src/main/scala/io/sorne/tlang/interpreter/type/Double.scala
package io.sorne.tlang.interpreter.`type`

import io.sorne.tlang.interpreter.Value

class Double(value: scala.Double) extends Value[scala.Double] {
  override def getValue: scala.Double = value

  override def getType: Predef.String = getClass.getName

  override def compareTo(value: Value[scala.Double]): Int = this.value.compareTo(value.getValue)
}

File: src/main/java/tlang/core/Double.java
package tlang.core;

import tlang.internal.ClassType;

public class Double implements Value {

    public static final Type TYPE = ClassType.of(Double.class);

    private final double value;

    public Double(double value) {
        this.value = value;
    }

    public double get() {
        return value;
    }

    @Override
    public Double getValue() {
        return this;
    }

    @Override
    public Type getType() {
        return TYPE;
    }
}
